File: WebSocketChat.Frontend/src/app/shared/models/chat/messages/user-leave-message.ts
import { Message, MessageIdentifier } from '../message';
import { MessageType } from '../message-type';

export class UserLeaveMessage extends MessageIdentifier implements Message {
	
	messageType: MessageType = MessageType.UserLeaveMessage;

	/** The identifier of the room that the user has left. RoomIdentifier = 0 indicates that it was the server. */
	roomIdentifier: number;
}

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/Messages/UserLeaveMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat.Messages
{
    public class UserLeaveMessage : Message
    {
        public UserLeaveMessage()
        {
            MessageType = MessageType.UserLeaveMessage;
            Timestamp = DateTime.Now;
        }

        /// <summary>
        /// The identifier of the room - 0 = global room/server
        /// </summary>
        public int RoomIdentifier { get; set; }

        /// <summary>
        /// Specifies when the user did leave the server
        /// </summary>
        public DateTime Timestamp { get; set; }
    }
}
